internal/commands/permission.go

```diff
@@ -222,8 +223,16 @@ func checkCmdFunc(cmd *cobra.Command, args []string) error {
 	resp, err := client.CheckPermission(ctx, request, grpc.Trailer(&trailerMD))
 	if err != nil {
 		var debugInfo *v1.DebugInformation
-		if resp != nil {
-			debugInfo = resp.DebugTrace
+
+		// Check for the debug trace contained in the error details.
+		if errInfo, ok := grpcErrorInfoFrom(err); ok {
+			if encodedDebugInfo, ok := errInfo.Metadata["debug_trace_proto_text"]; ok {
+				debugInfo = &v1.DebugInformation{}
+				err = prototext.Unmarshal([]byte(encodedDebugInfo), debugInfo)
+				if err != nil {
+					return err
+				}
+			}
```

Comment:
Let's please add a test

Reply:
The main regression test is on the SpiceDB side but I'll see if I can add a test for parsing the proto here

Reply:
Added